doc/rst/tools/unstableWarningAnonymizer/unstableWarningAnonymizer.rst

```diff
@@ -0,0 +1,185 @@
+in your code, making sure that even if your code is not meant for the public,
+you can still share these important metrics with us.
+
+This analysis aids in understanding the distribution of unstable warnings,
+facilitating the identification and prioritization of common unstable features
+without compromising program confidentiality.
+
+For example, given the following output from a Chapel program:
+
+.. code-block:: sh
+
+  ra-unordered-atomics.chpl:75: warning: 'dmapped' keyword is unstable, instead please use factory functions when available
+  ra-unordered-atomics.chpl:76: warning: 'dmapped' keyword is unstable, instead please use factory functions when available
+  ../HPCCProblemSize.chpl:9: warning: The 'MemDiagnostics' module is unstable
+
+The script will generate output similar to:
+
+.. code-block:: sh
+
+  2 instances of "warning: 'dmapped' keyword is unstable, instead please use factory functions when available"
+  1 instance of "warning: The 'MemDiagnostics' module is unstable"
+
+The script can also generate csv output, include file counts, and more.
+See the help message for more details.
+
+Prerequisites
+-------------
+
+This tool has the following prerequisites:
+
+- A Unix-like environment (something with commands like cd, mkdir, rm, etc.)
+- Chapel built with Regular Expression support. (see how to enable it below)
+
+Enabling Regular Expression Support
+-----------------------------------
+
+Setting the environment variable CHPL_RE2 to bundled will enable regular
+expression support with the RE2 library:
+
+.. code-block:: sh
+
+   export CHPL_RE2=bundled
+
+Then, rebuild Chapel. The RE2 library will be expanded from a release included
+in the Chapel distribution.
+
+Building
+--------
+
+The script can be found under /tools/unstableWarningAnonymizer/unstableAnonScript.chpl
+Compile the script with the following command:
+
+.. code-block:: sh
+
+  chpl unstableAnonScript.chpl
+
+This should create an executable named `unstableAnonScript`.
+
+Usage
+-----
+
+The usage of the script is split into two stages:
+
+  1. Collecting output from your program's compilation and execution
+  2. Using the script to anonymize this output.
+
+
+Collecting the Output
+~~~~~~~~~~~~~~~~~~~~~
+
+Once the script is built, compile and run your program as you would normally,
+with the addition of the compilation flag `--warn-unstable` and save the full
+output into a file.
+`--warn-unstable` will cause any use of unstable features to trigger a
+warning, which will then be saved in the output file specified above.
+
+.. code-block:: sh
+
+  chpl --warn-unstable myProgram.chpl >myUnstableOutput.txt 2>&1 ./myProgram >>myUnstableOutput.txt 2>&1
+
+After these commands, `myUnstableOutput.txt` (or whatever you've named
+it) should contain any unstable warnings you may trigger in your code,
+as well as any other potential output that occurs when compiling and
+running your program.
```

Comment:
I think this command line won't execute `./myProgram`; I think it'll pass `./myProgram` as an extra argument to the compiler, e.g,

```
fortytwo@enodia:~/src/chapel (main)$ chpl foo.chpl > tmptmp 2>&1 ./foo >> tmptmp 2>&1
fortytwo@enodia:~/src/chapel (main)$ cat tmptmp 
error: file './foo' does not have a recognized suffix
```

I think you want 

```
chpl --warn-unstable myProgram.chpl > myUnstableOutput.txt 2>&1 && ./myProgram >> myUnstableOutput.txt 2>&1
```

(Note the `&&`.  Also it's more conventional to put a space between the `>` and its filename.)

Reply:
Thanks for catching that Paul! I've pushed a fix.